### ISSUE TITLE
Confusion over how event handlers work

### ISSUE THREAD

**Issue by David-Else** [Polymer/lit-html#757]
I can't get the event handler to work, I have read the docs.

https://lit-html.polymer-project.org/guide/writing-templates

I have tried the following with and without the `this._handleClick = this._handleClick.bind(this);` and the click handler is showing up in Chrome Dev tools with `handler: (e) this.handleEvent (e)` NOT pointing to the `_handleClick `function. I don't see what I am doing wrong, the click button does nothing.

The view is injected in the controller and called from there, the template renders fine. Is there some strangeness with the `.this` or am I doing something wrong?

```
import { html, render } from '../../node_modules/lit-html/lit-html.js';

export default class View  {
  constructor(element) {
    this.element = element;
    // this._handleClick = this._handleClick.bind(this);
  }

  _handleClick(e) {
    console.log('click');
  }

  renderAndUpdate(items) {
    const filesTemplate = items =>
      html`
        <button @click=${this._handleClick}>Click Me</button>
        <ul>
          ${
            items.map(
              item => html`
                <li>
                  Name:<strong>${item.fileName}</strong> SR:<strong
                    >${item.sampleRate}</strong
                  >
                </li>
              `,
            )
          }
        </ul>
      `;

    render(filesTemplate(items), this.element);
  }
}

const view = new View(document.getElementById('app'));
```

**Comment by fopsdev** [comment]
Hi David

I had to:
```
_handleClick = e => {
    this.actions.changeFirstName1()
  }

```
to make it work.

**Comment by b-strauss** [comment]
Your example doesn't render anything.

Adding `view.renderAndUpdate([]);` at the end makes it work.

https://codepen.io/anon/pen/gqOymP?editors=1010

**Comment by David-Else** [comment]
THANKS everyone!

I apologise, my code example was not complete and should have been runnable by itself. I thought it would be something so obvious it did not need running.

After seeing the example work in code pen I realized that it must be something else in the code, and guess what was in the CSS?!
```
.box * {
  pointer-events: none;
}
```
So total cringe on my behalf!

Lit-Html is amazing, eagerly awaiting the v1.0 release and the buzz I am sure it is going to generate.

**Comment by b-strauss** [comment]
Everyone makes mistakes. 😉

**Comment by David-Else** [comment]
@jsilvermist @b-strauss Thanks, but I am still struggling. There seem to be no docs relating directly to CSS manipulation, which I assume is going to be a massive use case. Some examples would be so helpful.

Here is my CSS:
```
button {
  color: orange;
  background-color: green;
}
```
and template `(this.col = 'red';)`:
```
        <button
          .style.color=${this.col}
          @dragover=${e => (this.element.style.backgroundColor = 'pink')}
          @dragleave=${e => (this.element.style.backgroundColor = 'orange')}
        >
          Click Me
        </button>
```
As you said, I don't want `this.element.style.backgroundColor = 'pink'` I just want to refer directly to the element that I am working on, `<button>`, but how can I do that?

`@dragover=${e => (.style.backgroundColor = 'pink')}`

Does not work, but I though lit-html was meant to map the . to the element's property?

` .style.color=${this.col}` does nothing, and I have no idea why? 

**TLDR Basically, after all this experimenting I want to do the following!** The above is just me trying to understand things. With this CSS:
```
#app {
  background-color: #72a2bc;
  border: 8px dashed transparent;
  transition: background-color 0.2s, border-color 0.2s;
}

#app.over {
  background-color: #a2cee0;
  border-color: #72a2bc;
}
```
I want too add the class 'over' when my event handler has 'dragEnter' and remove it when it has 'dragLeave'. My previous working code just called these methods, I would like to improve this by using lit-html :)

```
 dragEnter(e) {
    this.view.element.className += ' over';
  }

  dragLeave(e) {
    this.view.element.className = this.deleteOver(this.view.element);
  }
```
THANKS SO MUCH!

**Comment by David-Else** [comment]
Hi @justinfagnani , sure, but I think the title of a new issue here would be the same 'Confusion over how event handlers work'. I am feeling a bit guilty asking for so much help here, I have also asked for help on Stack Overflow for this issue:

https://stackoverflow.com/questions/54332184/how-do-i-add-and-remove-classes-using-event-listeners-in-lit-html

but nobody seems to have the answer :( Could you humour me and answer my question here or stack overflow? I am getting a bit desperate!
